### PR TITLE
Onboard Load Balancer (part 1: implement creation payload helpers)

### PR DESCRIPTION
**Pull request by joaopalet**
- Adds resource schema and model
- Adds `toCreatePayload` and tests

### CODE REVIEW

File: stackit/internal/services/loadbalancer/loadbalancer/resource.go [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+	}
+
+	r.client = apiClient
+	tflog.Info(ctx, "Load Balancer client configured")
+}
+
+// Schema defines the schema for the resource.
+func (r *projectResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	descriptions := map[string]string{
+		"main":                 "Load Balancer project resource schema.",
```
Comment (hcsa73) [inline stackit/internal/services/loadbalancer/loadbalancer/resource.go:157]: Is this a project?

File: stackit/internal/services/loadbalancer/loadbalancer/resource.go [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+				Computed:    true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"network_id": schema.StringAttribute{
+							Description: descriptions["network_id"],
+							Optional:    true,
+							Computed:    true,
+							Validators: []validator.String{
+								validate.UUID(),
+								validate.NoSeparator(),
```
Comment (hcsa73) [inline stackit/internal/services/loadbalancer/loadbalancer/resource.go:267]: You don't need the `NoSeparator`. That's only used for fields that are used to build the internal ID (in this case, `project_id` and `name`). It's so it's guaranteed that, when we split `id` by the separator during the `ImportState` routine, we aren't doing anything we shouldn't

File: stackit/internal/services/loadbalancer/loadbalancer/resource.go [hunk 3]
```diff
@@ -0,0 +1,551 @@
...
+							Computed:    true,
+						},
+					},
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: descriptions["name"],
+				Optional:    true,
+				Computed:    true,
+				Validators: []validator.String{
```
Comment (hcsa73) [inline stackit/internal/services/loadbalancer/loadbalancer/resource.go:250]: Add `NoSeparator` validator (for reasons described in the `network_id` comment)

File: stackit/internal/services/loadbalancer/loadbalancer/resource.go [hunk 4]
```diff
@@ -0,0 +1,551 @@
...
+
+	return &networks
+}
+
+func toOptionsPayload(ctx context.Context, model *Model) (*loadbalancer.LoadBalancerOptions, error) {
+	var optionsModel = &Options{}
+	if !(model.Options.IsNull() || model.Options.IsUnknown()) {
+		diags := model.Options.As(ctx, optionsModel, basetypes.ObjectAsOptions{})
+		if diags.HasError() {
+			return nil, fmt.Errorf("%v", diags.Errors())
```
Comment (hcsa73) [inline stackit/internal/services/loadbalancer/loadbalancer/resource.go:488]: Use `core.DiagsToError` instead

File: stackit/internal/services/loadbalancer/loadbalancer/resource.go (hunk 1 above)
Comment (joaopalet) [reply stackit/internal/services/loadbalancer/loadbalancer/resource.go:157]: Copy paste leftover

**Review by hcsa73 — APPROVED** [review]

**Pull request merged**